.github/workflows/infracost.yml

```diff
@@ -0,0 +1,74 @@
+name: Infracost CI
+
+on:
+  pull_request:
+    paths:
+      - 'qhub/template/stages/02-infrastructure'
+      - '.github/workflows/infracost.yml'
+
+jobs:
+  infracost:
+    name: Run Infracost on QHub Infrastructure
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: true
+      matrix:
+        provider:
+          - aws
+          - gcp
+          - azure
+
+    env:
+      TF_ROOT: qhub/template/stages/02-infrastructure/${{ matrix.provider }}
+
+    steps:
+      # Checkout the base branch of the pull request
+      - name: Checkout infrastructure
+        uses: actions/checkout@v3
+        with:
+          ref: '${{ github.event.pull_request.base.ref }}'
+
+      # Setup Infracost
+      - name: Setup Infracost
+        uses: infracost/actions/setup@v2
+        with:
+          api-key: ${{ secrets.INFRACOST_API_KEY }}
+
+      # Run Infracost on the base branch to generate the baseline cost
+      - name: Generate Infracost cost estimate baseline
+        run: |
+          infracost breakdown --path=${TF_ROOT} \
+                              --format=json \
+                              --out-file=/tmp/infracost-base.json
+
+      # Checkout the PR branch
+      - name: Checkout PR branch
+        uses: actions/checkout@v3
+
+      # Compute if any specific file has been changed in the specific cloud-provider Terraform files
+      - name: Verify Changed files
+        uses: tj-actions/verify-changed-files@v9.1
+        id: verify-changed-files
+        with:
+          files: |
+            qhub/template/stages/02-infrastructure/${{ matrix.provider }}
+
+      # Generate Infracost diff cost estimate only if the file has been changed
+      - name: Generate Infracost diff
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+        run: |
+          infracost diff --path=${TF_ROOT} \
+                          --format=json \
+                          --compare-to=/tmp/infracost-base.json \
+                          --out-file=/tmp/infracost.json
+
+      # Generate the cost estimate report and submit it as a PR comment only if the file has been changed
+      - name: Post Infracost comment
```

Comment:
Do we need to manually generate this file for it to start posting updates? I don't see any `infracost.json` or artifacts from the CI.

Reply:
It's because we are not saving it as an artifact yet, it's present in the machine env when it's run. Not sure if we would like to have this public information - I would suggest moving this action to integration tests...

Reply:
Yes, it does.